opencog/cython/opencog/utilities.pyx

```diff
@@ -52,3 +53,61 @@ def tmp_atomspace():
     finally:
         set_default_atomspace(parent_atomspace)
 
+
+def add_link(Type t, outgoing, TruthValue tv=None):
+    # create temporary cpp vector
+    cdef vector[cHandle] handle_vector
+    for atom in outgoing:
+        if isinstance(atom, Atom):
+            handle_vector.push_back(deref((<Atom>(atom)).handle))
+        else:
+            raise TypeError("outgoing set should contain atoms, got {0} instead".format(type(atom)))
+    cdef cHandle result
+    result = c_add_link(t, handle_vector)
+    if result == result.UNDEFINED: return None
+    atom = Atom.createAtom(result);
+    if tv is not None:
+        atom.tv = tv
+    return atom
+
+
+def add_node(Type t, atom_name, TruthValue tv=None):
+    """
+    Add Node to the atomspace from the current context
+    """
+    cdef string name = atom_name.encode('UTF-8')
+    cdef cHandle result = c_add_node(t, name)
+
+    if result == result.UNDEFINED: return None
+    atom = Atom.createAtom(result);
```

Comment:
create_python_value_from_c_value should be used instead, otherwise it returns incorrect Python wrapper

Reply:
create_python_value_from_c_value doesn't accept cHandle